### DISCUSSION TITLE
Correct way of normalizing the node features

### DISCUSSION THREAD

**Discussion by SaffronWolf** [pyg-team/pytorch_geometric#4381] [Q&A] [answered]
I am working on graph classification problem and was wondering how to normalize node features. Are they supposed to be normalized individually or for entire graph at a time? Also, how  do you ensure the same behavior at inference?

**Comment by rusty1s** [comment, marked as answer]
You should calculate mean and standard deviation of node features across all nodes in all training graphs:
```python
mean = train_dataset.data.x.mean(dim=0, keepdim=True)
std = train_dataset.data.x.std(dim=0, keepdim=True)
```
You can then apply normalization within every graph:
```python
data.x = (data.x - mean) / std
```

**Reply by SaffronWolf** [reply]
Thank you so much. Doing this gave me a freaking 50% jump in accuracy !!!!

**Reply by rusty1s** [reply]
Awesome :)
